### PR TITLE
Add usertype recovery to the menu to display optional link (e-learning)

### PR DESCRIPTION
**Pull request by Wardormeur**

### CODE REVIEW

File: cd-menu/cd-menu.js [hunk 1]
```diff
@@ -134,7 +135,24 @@ window.cdMenu = function (options) {
         loginRegister.style.display = 'block';
       });
     }
-
+    function canSeeELearningModule (userDojos) {
```
Comment (butlerx) [inline cd-menu/cd-menu.js:138]: could this not be more generic?

Comment (Wardormeur) [reply cd-menu/cd-menu.js:138]: We could, but until we don't have more use case, there is no point in generalizing? @DanielBrierton

**Comment by DanielBrierton** [comment]
Do we want to show the menu, then check if we have access to the e-learning module and show that if it's available, or wait until we know everything before showing the menu? At the moment with this PR, it's the former.

File: cd-menu/cd-menu.js (hunk 1 above)
Comment (DanielBrierton) [reply cd-menu/cd-menu.js:138]: Only other option I see is to have two separate functions isMentor, and isChampion, but then we're adding unnecessary loops. I'm ok with this until we have another use case that makes sense to generalise for.

**Comment by Wardormeur** [comment]
Considering we hide it afterwards (like cdf) I believe it's the former? Elsewhat it shoud be a deactivated item, not a hidden one ?

**Pull request merged**